internal/internal_workflow.go

```diff
@@ -158,6 +160,7 @@ type (
 		closed       atomic.Bool      // indicates that owning coroutine has finished execution
 		blocked      atomic.Bool
 		panicError   error // non nil if coroutine had unhandled panic
+		state        ExecutionState
```

Comment:
Thinking about this generally, why does a coroutine need a state? Shouldn't the only thing that matters be what state the workflow is in at that time? I could understand if this were for general purpose use, but it's only for readonly.

Maybe, stepping back, we should just have `commandsDisallowed atomic.Bool` on `dispatcherImpl` and keep it simple.

Reply:
I did consider that approach. I did the more complicated approach because it is minimal addition complexity and will be needed in the future to support things like local activities in update validators, that we have said we would like to prioritize.

Reply:
I think "local activities in queries/validators" should be solved when we get there, because I think internally the local activities will have to block and not let other workflow code resume like regular local activities might. And I don't think local activities called from queries/validators will produce commands anyways. They are just a way to make external calls. 

Arguably we could just let queries/validators do whatever non-deterministic things they want so long as they don't mutate workflow state, but I suspect we'll call them "local activities" just to make the separation of the concept clear.

So I still think a `commandsDisallowed` simple approach will work all around well.

Reply:
OK I am fine with that approach, I won't call it `commandsDisallowed` because in the Go SDK it is more then just `commandsDisallowed`, I am also blocking things like setting handlers or reading from channels.

Reply:
:+1: Yeah, "read only mode" or something